nncf/torch/pruning/base_algo.py

```diff
@@ -111,27 +112,59 @@ def _prune_weights(self, target_model: NNCFNetwork):
                 assert self._is_pruned_module(module)
 
                 nncf_logger.info("Adding Weight Pruner in scope: {}".format(node_name))
-                operation = self.create_weight_pruning_operation(module)
-                hook = operation.to(device)
+                pruning_block = self.create_weight_pruning_operation(module, node_name)
+                # Hook for weights and bias
+                hook = UpdateWeightAndBias(pruning_block).to(device)
                 insertion_commands.append(
                     PTInsertionCommand(
-                        PTTargetPoint(TargetType.OPERATION_WITH_WEIGHTS,
+                        PTTargetPoint(TargetType.PRE_LAYER_OPERATION,
                                       target_node_name=node_name),
                         hook,
                         TransformationPriority.PRUNING_PRIORITY
                     )
                 )
-
                 group_minfos.append(PrunedModuleInfo(node_name=node_name,
                                                      module_scope=module_scope,
                                                      module=module,
-                                                     operand=hook,
+                                                     operand=pruning_block,
                                                      node_id=node.node_id))
+
             cluster = Cluster[PrunedModuleInfo](i, group_minfos, [n.node_id for n in group.elements])
             self.pruned_module_groups_info.add_cluster(cluster)
+
+        # Propagate masks to find norm layers to prune
+        init_output_masks_in_graph(target_model_graph, self.pruned_module_groups_info.get_all_nodes())
+        MaskPropagationAlgorithm(target_model_graph, PT_PRUNING_OPERATOR_METATYPES).mask_propagation()
+
+        # Adding binary masks also for Batch/Group Norms to allow applying masks after propagation
+        types_to_apply_mask = ['group_norm']
+        if self.prune_batch_norms:
+            types_to_apply_mask.append('batch_norm')
+
+        all_norm_layers = target_model_graph.get_nodes_by_types(types_to_apply_mask)
+        for node in all_norm_layers:
+            if node.data['output_mask'] is None:
+                # Skip elements that will not be pruned
+                continue
+
+            node_name = node.node_name
+            module = target_model.get_containing_module(node_name)
+
+            pruning_block = self.create_weight_pruning_operation(module, node_name)
```

Comment:
I wonder why norm layers are handled separately from clustered groups of layers to be pruned?
Why not to handle them with all pruned layers in `create_pruning_groups` method?